File: Msys_Programs/MSys_Projects/GoProg/DataStructure/Binary_Search/Binary_Search_Recursive.go
package main

import "fmt"

func binaryrecursive(A []int, key, l, r int) int {
	if l > r {
		return -1
	} else {
		mid := (l + r) / 2
		if key == A[mid] {
			return mid
		} else if key < A[mid] {
			return binaryrecursive(A, key, l, mid-1)
		} else if key > A[mid] {
			return binaryrecursive(A, key, mid+1, r)
		}
	}
	return 0
}

func main() {
	A := []int{25, 36, 65, 77, 88, 90, 96, 99}
	b := len(A)
	// f := binaryrecursive(A, 96, 0, b)
	// fmt.Println("result:", f)
	fmt.Println("result:", binaryrecursive(A, 96, 0, b))

}

File: Msys_Programs/MSys_Projects/PythonProg/Python_DSA/Binary_Search/Binary_Search_Recursive.py
def binarysearch_recursive(A,key,l,r):
    if l > r:
        return -1
    else:
        mid = (l + r) // 2
        if key == A[mid]:
            return mid
        elif key < A[mid]:
            return binarysearch_recursive(A, key, l, mid-1)
        elif key > A[mid]:
            return binarysearch_recursive(A, key, mid+1, r)


A = [25,36,38,44,56,66,68,70]
# f = binarysearch_recursive(A, 55, 0, 7)
f = binarysearch_recursive(A, 68, 0, 7)
print("result", f)